File: Taller1/Java/src/Taller1.java
import java.util.Scanner;



public class Taller1 {
	
	public static void main(String[] args) {
	    Scanner sc = new Scanner(System.in);

		System.out.println("Escriba el limite superior");
	    int limite = sc.nextInt();

	    System.out.println("Escriba el tiempo del sleep");
	    int sleep = sc.nextInt();
	    
	    
	    Scanner s = new Scanner(System.in);

	    System.out.println("Escriba 't' para usar la extensión de clase Thread o 'r' para la interface Runnable");
	    String c = s.nextLine();
	    
	    switch(c) {
	    case "r":
			Thread tr = new Thread(new ImparRunnable(limite, sleep));
			Thread pr = new Thread(new ParRunnable(limite, sleep));
			
			tr.start();
			pr.start();
			break;
	    case "t":
			ParThread tp = new ParThread(limite, sleep);
			ImparThread t = new ImparThread(limite, sleep);

			t.start();
			tp.start();
			break;
	    default:
		    System.out.println("Comando no valido");
		    break;
	    }
	    
	    

	    

		

		
	}
	
}

File: Taller1/Python/Taller1.py
import threading
import time


def esPar(n):
    if n % 2 == 0:
        return True
    return False

def imprimirPares (limite, dormir) :
    time.sleep(int(dormir) / 1000)
    for i in range(int(limite)):
        if esPar(i):
            print("ESPAR: " + str(i))

def imprimirImpares (limite, dormir):
    time.sleep(int(dormir) / 1000)
    for i in range(int(limite)):
        if not esPar(i):
            print("NOESPAR: " + str(i))


limite = input("Escriba el limite superior ")
dormir = input("Escriba el tiempo que los  threads van a dormir ")


thread1 = threading.Thread(target=imprimirImpares, args=(limite, dormir))
thread2 = threading.Thread(target=imprimirPares, args=(limite, dormir))

thread1.start()
thread2.start()